### PR TITLE
[BIT] add feature place, fix block_multihead_attention

### PR DESCRIPTION
**Pull request by Cutelemon6**
### block_multihead_attention
根据示例对 paddle.incubate.nn.functional.block_multihead_attention 的诸多 tensor 初始化进行了限制，参照 api 文档的[代码示例](https://www.paddlepaddle.org.cn/documentation/docs/en/develop/api/paddle/incubate/nn/functional/masked_multihead_attention_en.html#masked-multihead-attention) 对 padding_offset, cum_offsets, cu_seqlens_q, cu_seqlens_k 进行了初始化，其它 tensor 有一些数值上的约束，这里不多赘述。

其中在初始化参数位置 22: max_dec_len_this_time 和 21: max_enc_len_this_time 遇到了要求这两个 tensor 需要放置在 cpu 上，查阅源码发现 TensorConfig 初始化 paddle tensor 的时候并没有显式指定 place，为此
1. 在 TensorConfig 中添加 self.place 成员，在其中的 \_\_init__ 方法添加缺省参数 place=None，并进行对应的赋值；在 \_\_deepcopy__ 中添加对应的深拷贝语句
2. TensorConfig 并在 config_analyzer.py 中的 get_numpy_tensor 中初始化上述两个位置参数期间设置 self.place 为 cpu，并在下一个位置参数初始化时设置 self.place 为 gpu

### CODE REVIEW

File: tester/api_config/config_analyzer.py [hunk 1]
```diff
@@ -36,9 +36,10 @@
 ]
 
 class TensorConfig:
-    def __init__(self, shape, dtype):
+    def __init__(self, shape, dtype, place=None):
         self.shape = shape
         self.dtype = dtype
+        self.place = place
```
Comment (wanghuancoder) [inline tester/api_config/config_analyzer.py:42]: 测试CPU可以有两种方式：
1. 安装CPU的最新develop包
2. 在程序执行最开始通过paddle.device.set_device("cpu")设置Paddle工作在CPU模式

**Review by wanghuancoder — APPROVED** [review]
